From 3b199e17e0d6a79e97c36021ec18ab7d7c58cd53 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Mon, 1 Sep 2014 13:10:38 -0400
Subject: [crypto] drbg: HMAC-SHA1 DRBG has crypto strength of 128 bits

Message-id: <E1XORNW-00088x-Q6@gondolin.me.apana.org.au>
Patchwork-id: 90073
O-Subject: [PATCH 16/19] crypto: drbg - HMAC-SHA1 DRBG has crypto strength of 128 bits
Bugzilla: 1011562
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Rafael Aquini <aquini@redhat.com>

RHEL7 Bugzilla #1011562

Upstream commit: 5b635e280efce5b735042c7ffa2487c4ba8fb4bf

The patch corrects the security strength of the HMAC-SHA1 DRBG to 128
bits. This strength defines the size of the seed required for the DRBG.
Thus, the patch lowers the seeding requirement from 256 bits to 128 bits
for HMAC-SHA1.

Signed-off-by: Stephan Mueller <smueller@chronox.de>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/crypto/drbg.c b/crypto/drbg.c
index 83cdd6f..1fcf322 100644
--- a/crypto/drbg.c
+++ b/crypto/drbg.c
@@ -184,7 +184,7 @@ static const struct drbg_core drbg_cores[] = {
 #endif /* CONFIG_CRYPTO_DRBG_HASH */
 #ifdef CONFIG_CRYPTO_DRBG_HMAC
  {
-  .flags = DRBG_HMAC | DRBG_STRENGTH256,
+  .flags = DRBG_HMAC | DRBG_STRENGTH128,
   .statelen = 20, /* block length of cipher */
   .max_addtllen = 35,
   .max_bits = 19,
-- 
1.7.1